execution_engine/src/system/auction/detail.rs

```diff
@@ -483,3 +498,20 @@ pub(crate) fn era_validators_from_snapshot(
         })
         .collect()
 }
+
+fn is_under_max_delegator_cap(
+    max_delegators_per_validator: Option<u32>,
+    new_validator_delegator_len: usize,
+) -> bool {
+    if max_delegators_per_validator.is_none() {
+        return true;
+    }
+
+    if let Some(max_delegators) = max_delegators_per_validator {
+        if new_validator_delegator_len >= max_delegators as usize {
+            return false;
+        }
+    }
+
+    true
```

Comment:
A little more concise:
```suggestion
    max_delegators_per_validator
        .map(|limit| new_validator_delegator_len < limit as usize)
        .unwrap_or(true)
```

Reply:
Applied suggestion